### PR TITLE
Add `base_url` documentation to SDK usages

### PR DESCRIPTION
**Pull request by nathanthorpe**
- Also error out if it can't determine the base URL

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,10 @@ or you can install the main branch of the repo by running:
 
 ## Authentication
 
-Upon first use, the Cirro client will ask if you would like to save your login information and give you a link to authenticate through the web browser.
+Upon first use, the Cirro client will ask if you what Cirro instance to use and if you would like to save your login information.
```
Comment (shanosborne) [inline README.md:22]: "the Cirro client will ask you what Cirro instance" - delete "if"

File: samples/Getting_started.ipynb [hunk 2]
```diff
@@ -38,9 +38,13 @@
     }
    },
    "source": [
-    "### Authenticating with your Data Portal log in\n",
+    "### Authenticating with Cirro\n",
     "\n",
-    "Before you use the Cirro CLI, you will need to authenticate yourself. You can do this by running the command below. If the package cannot already find your information (i.e. you or your machine do not have credentials already saved), then this command it will give you a link to log in through a browser window, the same way you would log into the Data Portal website."
+    "Before you use the Cirro CLI, you will need to authenticate yourself. You can do this by running the command below. If the package cannot already find your information (i.e., you do not have credentials already saved), then this command will give you a link to log in through a browser window, the same way you would log into the Data Portal website.\n",
+    "\n",
+    "You will also need to set the base URL of the Cirro instance you wish to access. To find it, log in to Cirro and check the URL in your browser.\n",
```
Comment (shanosborne) [inline samples/Getting_started.ipynb:45]: Could we put the example of 'app.cirro.bio' in this line too? I know its in a lot of places, I just want to be clear what they're looking for when we say "check the url"

File: cirro/config.py [hunk 3]
```diff
@@ -81,8 +82,12 @@ def __init__(self, base_url: str = None):
         self.user_config = load_user_config()
         self.base_url = (base_url or
                          os.environ.get('CIRRO_BASE_URL') or
-                         (self.user_config.base_url if self.user_config else None) or
-                         Constants.default_base_url)
+                         (self.user_config.base_url if self.user_config else None))
+        if not self.base_url:
+            raise RuntimeError('No base URL provided, please provide the base_url parameter,'
+                               ' set the CIRRO_BASE_URL environment variable, or run cirro configure')
```
Comment (shanosborne) [inline cirro/config.py:88]: How do you feel about switching the order of these options in the error msg to be cirro configure, then env variable, and then parameter?

Comment (nathanthorpe) [reply cirro/config.py]: Yeah, will update that one also.

**Review by shanosborne — APPROVED** [review]

**Pull request merged**